cfme/utils/template/base.py

```diff
@@ -125,9 +125,18 @@ def raw_image_url(self):
             raise TemplateUploadException("Cannot get image URL.")
         else:
             image_name = self.image_pattern.findall(string_from_url)
-            if len(image_name):
+            if len(image_name) == 1:
                 # return 0th element of image_name, likely multiple formats for same provider type
                 return '/'.join([self.image_url, image_name[0]])  # TODO support multiple formats
+            elif len(image_name) > 1:
+                if 'manageiq' in self.image_url:  # Upstream ovirt build
+                    index = (i for i, string in enumerate(image_name) if 'ovirt' in string
```

Comment:
Required:
No need to find an index for a particular image. I would like to go with
`image_name = [img for img in image if 'ovirt' in img  and 'qc2' in img][0]`

Reply:
So I think any checks like this that specific to a provider type, should be handled in the respective provider's upload class.

This is addressing the TODO on L130, handling multiple formats.

I'm considering turning raw_image_url into a method that takes a parameter of a callback for determining which format to choose.

This can also be handled in the template_upload script itself, and iterating over all raw_image_urls returned by the method.

Reply:
@digitronik , I have updated the regex in cfme/utils/template/rhevm.py to match *rhevm*.qcow2 or *ovirt*.qc2 files,
so the changes in cfme/utils/template/base.py are not needed anymore for RHV/Ovirt.

However, we still have to add support for multiple image formats for same provider type.I will be doing that in a follow up PR.